HtmlEditor: add Overview topic + small fixes (#5168)
DevExpress/devextreme-documentation: concepts/05 UI Components/HtmlEditor/05 Overview.md

> HtmlEditor is a client-side WYSIWYG editor that allows users to format text and visual content and export it as HTML or Markdown. Supported features include:

Reviewer: ```suggestion
HtmlEditor is a client-side WYSIWYG editor that allows users to format text and visual content, and to export it as HTML or Markdown. Supported features include:
```

> HtmlEditor is a client-side WYSIWYG editor that allows users to format textual and visual content and output it as HTML or Markdown. Supported features include:

Reviewer 2: > textual

"text" in form of an adjective is just "text".

> output it

What do you mean by "output"? Export? Save?

```suggestion
HtmlEditor is a client-side WYSIWYG editor that allows users to format text and visual content and export it as HTML or Markdown. Supported features include:
```

> - [Inline formats](/concepts/05%20UI%20Components/HtmlEditor/10%20Formats '/Documentation/Guide/UI_Components/HtmlEditor/Formats/'):

Reviewer 2: Just a side note -- shouldn't you capitalize these topic names? The same comment applies all similar occurrences.

Reviewer 2: Well, I believe we need to decide what to do with capitalization.

Reviewer 2: Everything is done. The issue is resolved.

>     - **Bold**, *italic*, <s>strikethrough</s> text formatting

Reviewer 2: Does it support underline?

```suggestion
    - **Bold**, *italic*, and <s>strikethrough</s> text formatting attributes
```

>     - Typeface, font size, text color changes (HTML only)

Reviewer 2: ```suggestion
    - Typeface, font size, and text color changes (HTML only)
```

- [Block formats](/concepts/05%20UI%20Components/HtmlEditor/10%20Formats '/Documentation/Guide/UI_Components/HtmlEditor/Formats/'):
    - Headers
    - Lists (ordered and unordered)
    - Code blocks
    - Quotes
- [Built-in format customization](/concepts/05%20UI%20Components/HtmlEditor/10%20Formats/33%20Customize%20Built-In%20Formats%20and%20Modules '/Documentation/Guide/UI_Components/HtmlEditor/Formats/#Customize_Built-In_Formats_and_Modules')
- [HTML and Markdown support](/concepts/05%20UI%20Components/HtmlEditor/00%20Getting%20Started%20with%20HtmlEditor/10%20Set%20the%20Output%20Markup%20Language.md '/Documentation/Guide/UI_Components/HtmlEditor/Getting_Started_with_HtmlEditor/#Set_the_Output_Markup_Language')

> - [Mail Merge](/api-reference/10%20UI%20Components/dxHtmlEditor/1%20Configuration/variables '/Documentation/ApiReference/UI_Components/dxHtmlEditor/Configuration/variables/')

Reviewer 2: Right here you capitalized the title. If you take a look at all list entries here, your capitalization is inconsistent.

Author: It was a mistake, I wanted them all uncapitalized :)
Also I have a question regarding dots at the end of the list items - I felt a bit lost at where should I put them and where I shouldn't. Is everything okay regarding dots at the end of sentences?

Reviewer 2: OK, I see. I thought as much.

Regarding periods at the end of sentences... It's bit more tricky than it seems. First of all, I don't have a preview for this pull request, therefore, I don't know how all this will look like in the resulting documentation.

In general, I would do as follows:

1. I would not put periods at the end of lines if they consist of links to other topics.\
2. I would put periods at the end of ordinary text lines.

Either way, it's difficult for me to imagine how this topic would look like. I'll mark this issue as resolved.

> - [Adaptive toolbar](/concepts/05%20UI%20Components/HtmlEditor/00%20Getting%20Started%20with%20HtmlEditor/20%20Configure%20the%20Toolbar.md '/Documentation/Guide/UI_Components/HtmlEditor/Getting_Started_with_HtmlEditor/#Configure_the_Toolbar') for working with images, links, and color formats

Reviewer 2: ```suggestion
- [Adaptive toolbar](/concepts/05%20UI%20Components/HtmlEditor/00%20Getting%20Started%20with%20HtmlEditor/20%20Configure%20the%20Toolbar.md '/Documentation/Guide/UI_Components/HtmlEditor/Getting_Started_with_HtmlEditor/#Configure_the_Toolbar') designed to work with images, links, and color formats.
```

- Image upload: drag-and-drop images onto the form, select files from the file system, or specify a URL.

> - Copy-paste rich content (the control strips unsupported formats).

Reviewer 2: Well, "unsupported formats" speaks more of data formats, including file formats. In this context, it's clear that you've meant text formatting attributes like bold, italic, subscript/superscript, strikeout, etc.

```suggestion
- Copy-paste rich content (the control automatically removes unsupported text formatting attributes).
```

> - [Mentions](https://js.devexpress.com/Demos/WidgetsGallery/Demo/HtmlEditor/Mentions/) (for example, @person)

Reviewer 2: ```suggestion
- [Mentions](https://js.devexpress.com/Demos/WidgetsGallery/Demo/HtmlEditor/Mentions/) (@person, for example)
```

- [Tables](/concepts/05%20UI%20Components/HtmlEditor/00%20Getting%20Started%20with%20HtmlEditor/30%20Work%20with%20Tables.md '/Documentation/Guide/UI_Components/HtmlEditor/Getting_Started_with_HtmlEditor/#Work_with_Tables')

## How to Use HTMLEditor

> HtmlEditor is designed to create rich text and export this text to HTML or Markdown. You can also use the component to parse HTML (for example, if you set [value](/Documentation/ApiReference/UI_Components/dxHtmlEditor/Configuration/#value) to markup), but this approach may not be effective since HtmlEditor does not support all HTML features. Note the following:

Reviewer 2: Approach. Ray discouraged this word usage. I would have preferred "technique" instead.

I also made some adjustments to enhance wording as well as overall text flow:

```suggestion
HtmlEditor is designed to create rich text and export it in HTML or Markdown format. You can also use the component to parse HTML content (for example, if you set [value](/Documentation/ApiReference/UI_Components/dxHtmlEditor/Configuration/#value) to markup). However, this technique may prove ineffective because HtmlEditor does not support all HTML features. The following limitations apply:
```

- If you use <a href="https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM" target="_blank">Shadow DOM</a>, the HtmlEditor component may experience issues in some browsers (see <a href="https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot#browser_compatibility" target="_blank">getSelection</a>).

> - HtmlEditor does not produce a fully structured HTML document with a `<!DOCTYPE>`, a `<head>`, and a `<body>`. Its purpose is to output markup that contains formatted rich content for an article, forum post, and so on.

Reviewer 2: fully-structured is an adjective, therefore, it needs a hyphen.

> a `<!DOCTYPE>`, a `<head>`, and a `<body>

You enumerate named entities here. This is exactly one of the article cases from the new guideline.

> Its purpose

Whose purpose? The pronoun may point anywhere in your scenario.

```suggestion
- HtmlEditor does not produce a fully-structured HTML document with `<!DOCTYPE>`, `<head>`, and `<body>` tags. The editor is meant to export simple markup that contains formatted rich content for an article, forum post, etc.
```

> - HtmlEditor saves only a subset of [tags and attributes](/concepts/05%20UI%20Components/HtmlEditor/05%20Tags%20and%20Attributes.md '/Documentation/Guide/UI_Components/HtmlEditor/Tags_and_Attributes/'). Everything else is discarded.

Reviewer 2: ```suggestion
- HtmlEditor saves only a limited subset of [tags and attributes](/concepts/05%20UI%20Components/HtmlEditor/05%20Tags%20and%20Attributes.md '/Documentation/Guide/UI_Components/HtmlEditor/Tags_and_Attributes/'). All other formatting attributes and features are discarded.
```

> - HtmlEditor automatically reduces unnecessary tags:

Reviewer 2: Same as before.

```suggestion
- HtmlEditor automatically removes redundant tags:
```

        <!-- from -->
        <p><span>He</span><em><span>llo</span></em></p>

        <!-- to -->
        <p>He<em>llo</em></p>

> - The component is not designed to convert text that is enclosed into `{}` characters into a variable. Markup with such text is processed as plain text.

Reviewer 2: > Markup with such text is processed as plain text.

Could you please clarify what does it mean? Does it mean that the entire content is treated as plain text if the content includes text between curly brackets? Or the component merely treats only bracketed content as plain text?

My bet is on the latter.

```suggestion
- The component is not designed to convert text enclosed between curly brackets (`{}`) into a variable. Content between curly brackets is always treated as plain text.
```

> - The HtmlEditor tables are native HTML tables and use native features, which leads to limitations. For example, users cannot past multiline text in separate cells.

Reviewer 2: ```suggestion
- The HtmlEditor tables are native HTML tables and use native features, therefore, the same limitations apply. For example, users cannot past multiline text in separate cells.
```

> - The HtmlEditor tables are not designed to add complex elements to cells like block elements, lists, nested tables, etc.

Reviewer 2: ```suggestion
- The HtmlEditor tables do not support complex elements in cells, such as block elements, lists, nested tables, etc.
```

> - The HtmlEditor tables are native HTML tables and use native features, therefore, the same limitations apply. For example, users cannot past multiline text in separate cells.

Reviewer 2: ```suggestion
- The HtmlEditor tables are native HTML tables and use native features, therefore, the same limitations apply. For example, users cannot paste multiline text in separate cells.
```

> - The HtmlEditor tables are native HTML tables and use native features, therefore, the same limitations apply. For example, users cannot paste multiline text in separate cells.

Reviewer: ```suggestion
- The HtmlEditor tables are native HTML tables and use native features; therefore, the same limitations apply. For example, users cannot paste multiline text in separate cells. [same note about phrasing here]
```

> - HtmlEditor does not support pasting text from Microsoft Word.

Reviewer 2: ```suggestion
- HtmlEditor does not support the clipboard data format of Microsoft Word®.
```

P.S. Can you paste content from LibreOffice Writer? This entry looks very limited to me...

> - HtmlEditor does not support the complex clipboard data format of text processors like Microsoft Word®.

Reviewer 2: ```suggestion
- HtmlEditor does not support complex clipboard data formats of text processor applications, such as Microsoft Word®.
```
